scripts/migrate-yaml.py

```diff
@@ -0,0 +1,193 @@
+#!/usr/bin/env python
+# Copyright 2016, Rackspace US, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# (c) 2016, Matthew Thode <matt.thode@rackspace.com>
+
+"""
+Print out values that have changed, by default just additions and changes.
+
+Additions can be taken as is and placed into an overrides file, but should
+you should verify you still want to keep it.
+
+Removals are not printed by default because not having an override is the
+same as taking a default.
+
+Changes are printed, showing what you overrode and what is getting overridden.
+
+CAUTION: danger-mode is only if you KNOW what you are overriding doesn't
+    need to change, warrantee is void if used.
+"""
+
+from __future__ import print_function
+import argparse
+from collections import namedtuple
+from deepdiff import DeepDiff
+import errno
+import yaml
+
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--overrides', dest='overrides', required=True,
+                        help='The path to the file with overrides.')
+    parser.add_argument('--info', dest='info', type=bool, default=False,
+                        help='Shows what is NOT overridden.')
+    parser.add_argument('--danger-mode', dest='danger',
+                        type=bool, default=False,
+                        help='Print output usable as an overrides file.')
+    return parser.parse_args()
+
+
+def root_cleanup(var):
+    """Cleans up what deepdiff returns so we can have some cleaner output.
+
+    :param var: the key you wish to clean up, just a string
+    :return: the cleaned up string
+    """
+    prefix = "root['"
+    suffix = "']"
+    var = var[var.startswith(prefix) and len(prefix):]
```

Comment:
I just noticed this and I'm not thrilled about it. I know it works, but it's too clever and you're hiding things in the slice. It'd be more explicitly written using if statements. I don't know if other people agree.